### ISSUE TITLE
Performance problems when generating DDL on large schemas or many tables

### ISSUE THREAD

**Issue by cdabel** [dbeaver/dbeaver#8473]
#### System information: 
- MacOS Catalina 10.15.4
- DBeaver version 7.0.2 (Community)
- Additional extensions:  None

#### Connection specification:
- SQL Server 2016:
       -> DBeaver canned, standard drivers for SQL Server:
              - com.microsoft.sqlserver:mssql-jdbc:RELEASE [7.4.1.jre8]
              - com.microsoft.sqlserver:mssql-jdbc_auth:RELEASE [8.2.0.x64]
- Redshift:
       -> com.amazon.redshift:redshift-jdbc42:RELEASE

- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
  I use my company's VPN (OpenVPN); other than that, no SSH or anything like that.

#### Describe the problem you're observing:
Redshift performance issue:
----------------------------
Compared to DataGrip, the performance when generating DDL on many tables is very slow. I tested DataGrip for the same set of tables and the DDL generated very quickly (<5 seconds). 
DBeaver became non-responsive (with Mac's "rainbow wheel of death") when trying to Generate DDL for a large amount of tables (approximately 150 tables) on Redshift. I had to kill DBeaver after over 10 minutes of waiting. I tested selecting only 10 tables and it takes 30 seconds to complete successfully.
DataGrip is able to complete this same amount in under 1 second.

SQL Server user experience issues:
------------------------------------
When trying to Generate DDL at the schema level within SQL Server, the performance is good as long as there are no Views included. However, if the schema contains SQL View objects, then the request errors out after about 90 seconds with:

```SQL Error [15197] [S0001]: There is no text for object 'dbo.VW_<MyViewName>'.
  There is no text for object 'dbo.VW_<MyViewName>'.
  There is no text for object 'dbo.VW_<MyViewName>'.```

Acknowledge the error and note that the Generated SQL window is empty.

I also noticed that there is no Generate SQL/DDL right-click context menu when clicking on `Tables` parent folder in Advanced Connection View. I would expect that the Generate DDL option should appear here, but since it does not, you have to either multi-select all the tables (which isn't hard to do), or you have to right-click on the schema object (either in Simple or Advanced Connection view) and try to Generate DDL, but as explained above, this takes a minute and a half, then errors out.  

#### Steps to reproduce, if exist:
Steps detailed above.

Redshift:
 1. Select approximately 100-200 tables within a schema and Generate DDL for them. DBeaver becomes unresponsive and after 10 minutes of waiting. Note that it takes 30 seconds to generate DDL for only 10 tables. 

#### Include any warning/errors/backtraces from the logs
[dbeaverlog.log](https://github.com/dbeaver/dbeaver/files/4494528/dbeaverlog.log)

After looking through the logs, I'm not sure these errors are caught in the logs, but here are the logs from during my session. You can see in the logs where I killed Dbeaver and restarted it:
```!ENTRY org.eclipse.core.resources 2 10035 2020-04-17 11:19:37.444
!MESSAGE The workspace exited with unsaved changes in the previous session; refreshing workspace to recover changes.```

Log is attached.

**Comment by uslss** [comment]
thanks for the bug report

**Comment by uslss** [comment]
is it still actual for you?

**Comment by cdabel** [comment]
Yes, still happens. I just highlight 100 tables or so, and Generate DDL, and it happens.

**Comment by jwalker5006** [comment]
This happens to me whenever I try to connect to a remote MS SQL server. 

Meaning, I connect to the exact same SQL server, 
on the LAN, I can see and edit all objects.
on the WAN, I can see all objects, but editing views and stored procedures show the SQL Error [15197] [S0001]: There is no text for object error

Exact same username and password, exact same databases, schema and everything.

I don't know how a firewall could allow only stored procedures to not be edited. I can run the stored procedures just fine form both LAN and WAN.

**Comment by navy1978** [comment]
@jwalker5006 same issue here

With remote DB connection it's super super super slow... Did you find a workaround?
